src/components/Inventory/AddInventory.js

```diff
@@ -0,0 +1,331 @@
+import { useRef, useState, useEffect } from "react";
+import "./AddInventory.scss";
+import { useNavigate } from "react-router-dom";
+import axios from "axios";
+
+import BackIcon from "../../assets/icons/arrow_back-24px.svg";
+import ErrorIcon from "../../assets/icons/error-24px.svg";
+
+const AddInventory = () => {
+  const navigate = useNavigate();
+
+  const initialState = useRef({
+    item_name: "",
+    description: "",
+    category: "",
+    status: "In Stock", // default status
+    quantity: "",
+    warehouse_name: "",
+  });
+
+  // State to keep track of form data
+  const [formData, setFormData] = useState(initialState.current);
+
+  //state to keep track of user errors
+  const [errors, setErrors] = useState({});
+
+  // State to track valid form submissions
+  const [validSubmissionCount, setValidSubmissionCount] = useState(0);
+
+  // State to track the submission ready version of the form data
+  const [submissionData, setSubmissionData] = useState(null);
+
+  // States to fetch and track warehouses
+  const [warehouses, setWarehouses] = useState([]);
+  const [warehousesLoaded, setWarehousesLoaded] = useState(false);
+
+  // Function to handle change in form data
+  const handleChange = (e) => {
+    const { name, value } = e.target;
+    setFormData((prevFormData) => ({ ...prevFormData, [name]: value }));
+  };
+
+  // Function to Validate user inputs
+  const validateForm = (inputValues) => {
+    const errors = {};
+    const validations = {
+      item_name: "Item's name missing",
+      description: "Item's description missing",
+      category: "Item's category missing",
+      status: "Status missing",
+      warehouse_name: "Warehouse missing",
+    };
+    Object.keys(validations).forEach((key) => {
+      if (!inputValues[key]) {
+        errors[key] = validations[key];
+      }
+    });
+
+    // Custom validation to check if quantity is a number
+    if (inputValues.quantity === undefined || inputValues.quantity === "") {
+      errors.quantity = "Quantity missing";
+    } else if (
+      isNaN(Number(inputValues.quantity)) ||
+      Number(inputValues.quantity) < 0
+    ) {
+      errors.quantity = "Quantity must be a positive number";
+    }
+    return errors;
+  };
+
+  //Function to handle form submission
+  const handleSubmit = (e) => {
+    e.preventDefault();
+
+    // Adjust formData for Out of Stock items before validation and submission
+    const adjustedFormData = {
+      ...formData,
+      quantity: formData.status === "Out of Stock" ? "0" : formData.quantity,
+    };
+
+    const validationErrors = validateForm(adjustedFormData);
+    setErrors(validationErrors);
+    // If there are no errors, set shouldSubmit to true
+    if (Object.keys(validationErrors).length === 0) {
+      setSubmissionData(adjustedFormData); // Set adjusted data for submission
+      setValidSubmissionCount((count) => count + 1); // Trigger the useEffect
```

Comment:
An improvement I see here is that instead of triggering a useEffect you could just make the POST request here directly.

Also would be nice UX if after successful form submission, you redirect the user to the inventory list page to view the newly created inventory

Reply:
Thanks for the feedback Rajat! 
Just sharing my thought process for the redirection on form submission: I did think of it, but decided not to do it on submission because if the client wishes to add multiple inventory items, they will keep having to come back to the form. What I did instead was have them redirected to the Inventory list page when they click the cancel button on the form, as the ticket did not specify the functionality for that button.